repo2docker/buildpacks/nix/nix-shell-wrapper

```diff
@@ -8,8 +8,25 @@ _term() {
 
 trap _term SIGTERM
 
-echo "$*"
-nix-shell default.nix --command "$*" &
+# if there is a binder/ sub-directory it takes precedence
+# files outside it are ignored
+if [ -e ./binder ]; then
+  nixpath="./binder/default.nix";
+  if [ -f ./binder/start ]; then
+    chmod u+x ./binder/start
+    nix-shell $nixpath --command "./binder/start $*" &
```

Comment:
Using `$@` here which is what the internet recommends leads to errors when the command is something like `jupyter --ip=0.0.0.0 ...` as `nix-shell` picks that up as an argument to it instead of the command.

There are several issues on the nix repos discussing this and adding support for `--` to indicate "all arguments after this are for the command, not nix-shell" but it seems they have stalled/not yet produced an implementation.

Let's use this unless someone knows what deep shell quoting magic we need :-/

Reply:
I added this comment as a comment in the script so it's preserved.